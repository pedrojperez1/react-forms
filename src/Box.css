.Box {
    display: inline-block;
    margin: 10px;
}

.Box-box {
    border-radius: 5%;
    margin: 5px;
}