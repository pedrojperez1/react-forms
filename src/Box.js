import React from "react";
import "./Box.css";

const Box = ({id, backgroundColor, height, width, removeFromBoxList}) => {
    const handleRemove = () => {
        removeFromBoxList(id);
    }
    return (
        <div className="Box">
            <div className="Box-box" style={{backgroundColor, height, width}}></div>
            <button onClick={handleRemove}>X</button>
        </div>
    )
};

export default Box;