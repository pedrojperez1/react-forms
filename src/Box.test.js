import { render } from '@testing-library/react';
import Box from './Box';

it("should not go up in smoke when rendering", () => {
    render(<Box id={123} backgroundColor="red" height={100} width={100}/>);
});

it("should render the same each time", () => {
    const {asFragment} = render(<Box id={123} backgroundColor="red" height={100} width={100}/>);
    expect(asFragment()).toMatchSnapshot();
})