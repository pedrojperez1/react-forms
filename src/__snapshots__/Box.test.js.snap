// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the same each time 1`] = `
<DocumentFragment>
  <div
    class="Box"
  >
    <div
      class="Box-box"
      style="background-color: red; height: 100px; width: 100px;"
    />
    <button>
      X
    </button>
  </div>
</DocumentFragment>
`;
